Add more fields to search on

my8bird opened the pull request:
Adding tags, status, and deciders and searchable fields.  Also, renamed verbatim to body so that in the search bar the user can easy search the body of the ADR.

These searches are now available powered by the luna package.
 * body: Text from the body
 * deciders: Tony
* tags: database
* status: draft

thomvaill commented:
Hi @my8bird, nice work!
Thank you for this PR.
I am currently on vacations with no laptop. I'll have a look when I come back, by the end of August.

thomvaill commented on packages/web/src/lib/search/Search.ts:
```diff
@@ -67,25 +67,33 @@ export class Search {
         {
-          title: adr.title || "Untitled",
-          verbatim: adr.body.enhancedMdx // TODO: remove tags (https://github.com/thomvaill/log4brains/issues/5)
+           title: adr.title || "Untitled",
```
Typescript type `AdrForSearch` must be updated too.
You can run `yarn typescript` to ensure your code is valid.
I found out that the typescript checks were indeed not run by the CI pipeline... This is now fixed. Could you please rebase on `master` so the checks are executed?

thomvaill commented:
As I said in comments, there is a typescript error to fix.
Also, I think we should give users a hint on how to use the Lunr syntax.
What about improving the  "Type to start searching" message when you click on search bar?

llwt commented:
Would you be open to someone taking this up and fixing the typings issue? Having tag search would be nice for bigger repos of ADRs.

my8bird commented:
By all means sure.  I was just getting a starting point in.  I forgot about
this actually so thanks for keeping it moving.

On Wed, Jan 18, 2023, 04:06 Steven Nance ***@***.***> wrote:

> Would you be open to someone taking this up and fixing the typings issue?
> Having tag search would be nice for bigger repos of ADRs.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thomvaill/log4brains/pull/82#issuecomment-1386793285>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAGJPFZ56ZGCS74RBGZNP3WS66BFANCNFSM55WZO5PQ>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

my8bird committed:
Add additional fields which a user can use to search

Note: this skipped the lint-staged step because it airbnb's dependency
has a main field set in the package json and the step wont pass

my8bird commented:
I updated to master and fixed the typescript errors.  however, the lint-staged step would not pass for me due to a miss configured package json in the air bnb eslint dependency so I skipped that locally.

yuri1969 commented:
Hi @thomvaill,
running `yarn install --frozen-lockfile && yarn build && yarn link-cli && yarn typescript` using ac9dfb0 works fine (on my machine).

Is there anything else left to do on this PR?

my8bird commented:
No every thing was working locally.  Might have been a bad version of that airbnb dep which was resolved.  If it meets your standards and is a good idea I would love to get it merged in

ms-ati commented:
@thomvaill are you able to merge this? Tag search is the motivation for my question! Also thank you so much for an amazing tool!
